lib/console.js

```diff
@@ -249,6 +252,74 @@ Console.prototype.groupEnd = function groupEnd() {
     this[kGroupIndent].slice(0, this[kGroupIndent].length - 2);
 };
 
+const untablableTypes = [
+  'symbol', 'undefined', 'boolean', 'number', 'string', 'function',
+];
+
+const untablableValues = [
+  undefined, null, NaN, true, false, global, process,
+];
+
+const untablable = (value) =>
+  untablableValues.includes(value) ||
+  untablableTypes.includes(typeof value);
+
+function getTablableKeysOf(O) {
+  if (isSet(O))
+    return Array.from({ length: O.size }, (e, i) => i);
+  if (isSetIterator(O))
+    return previewSetIterator(O, 50);
+  if (isMap(O))
+    return [...O.keys()];
+  if (isMapIterator(O))
+    return previewMapIterator(O, 50);
+
+  return Object.keys(O);
+}
+
+Console.prototype.table = function(data, only) {
+  if (untablable(data))
+    return this.log(data);
+
+  const rows = getTablableKeysOf(data);
+  let columns = [];
+
+  for (const row of rows) {
+    const item = data[row];
+    if (!item || (typeof item !== 'object' && typeof item !== 'function'))
+      continue;
+    for (const key of getTablableKeysOf(data[row])) {
+      if (!columns.includes(key) || (only !== undefined && only.includes(key)))
+        columns.push(key);
+    }
+  }
+
+  const deep = columns.length !== 0;
+  if (!deep)
+    columns = ['Values'];
+
+  let output = `(index)\t${columns.join('\t')}`;
+
+  for (const row of rows) {
+    let r = row;
+    for (const column of columns) {
+      let item = data[row];
+      if (deep) {
+        if (isMap(item))
+          item = item.get(column);
+        else if (isSet(item))
+          item = Array.from(item.values())[column];
+        else
+          item = item[column];
+      }
+      r += `\t${item}`;
```

Comment:
This will always print `\tundefined` for `Map` and `Set` but it is possible to get the value from those.